### ISSUE TITLE
CatBoost own metrics for evaluation - cyntax

### ISSUE THREAD

**Issue by algomaschine** [catboost/catboost#2446]
https://youtu.be/usdEWSDisS0?t=3775  here Anna Veronica is saying that one can use own metric to evaluate model's performance.

**Notebook** https://github.com/catboost/catboost/blob/671611e0a1ae47c03e7b2b52c06bdd91aeb23a5c/catboost/tutorials/events/2019_pydata_london/pydata_london_2019.ipynb#L4

**Section:** Metric evaluation on a new dataset

Can you clarify what would be the syntax to define my own eval metric to measure performance? 

The original code from the tutorial goes like that:

metrics = model.eval_metrics(  # eval_metric is to test model on completely new data
    data=validation_pool,
    metrics=['Logloss','AUC'],
    ntree_start=0,
    ntree_end=0,
    eval_period=1, # how often is eval metric recalculated, 1=every iteration, 10 - every 10th iteration (faster for slow metrics like AUC)
    plot=True
)

So I'd like to my custom function called pnl to evaluate during training, but GPT says "CatBoost does not natively support custom functions within the eval_metrics method. However, a workaround can be to calculate the metric manually after each training iteration", and then it proposes this kind of code below.

`
model = CatBoostClassifier(
    iterations=50,
    learning_rate=0.5,
    custom_loss=['AUC', 'Accuracy', 'F1']
)

# Start training and store model at each iteration
for i in range(50): 
    model.fit(X_train, y_train, 
              eval_set=(X_test, y_test), 
              verbose=False,
              plot=False,
              init_model=model if i > 0 else None)  # init_model helps in continuing training from the previous state

    # Predict after each iteration
    y_pred = model.predict(X_test)

    # Calculate PnL
    pnl = calculate_pnl(y_true, y_pred, prices)

    # Print or store PnL
    print(f"Iteration {i+1}: PnL = {pnl}")
`

Is there perhaps some more elegant solution from CatBoosts intrinsic capabilities as Anna was referring to in the video?

Thanks in advance!
